server/datastore/mysql/apple_mdm.go

```diff
@@ -1164,3 +1164,81 @@ func (ds *Datastore) InsertMDMIdPAccount(ctx context.Context, account *fleet.MDM
 	_, err := ds.writer.ExecContext(ctx, stmt, account.UUID, account.Username, account.Salt, account.Entropy, account.Iterations)
 	return ctxerr.Wrap(ctx, err, "creating new MDM IdP account")
 }
+
+func (ds *Datastore) GetMDMAppleFileVaultSummary(ctx context.Context, teamID *uint) (*fleet.MDMAppleFileVaultSummary, error) {
+	sqlFmt := `
+	SELECT
+	COUNT(
+		CASE WHEN EXISTS (
+			SELECT
+				1 FROM host_mdm_apple_profiles hmap
+			WHERE
+				h.uuid = hmap.host_uuid
+				AND hdek.decryptable = 1
+				AND hmap.profile_identifier = 'com.fleetdm.fleet.mdm.filevault'
+				AND hmap.status = 'applied'
+				AND hmap.operation_type = 'install') THEN
+			1
+		END) AS applied, COUNT(
+		CASE WHEN EXISTS (
+			SELECT
+				1 FROM host_mdm_apple_profiles hmap
+			WHERE
+				h.uuid = hmap.host_uuid
+				AND(hdek.decryptable = 0
+					OR hdek.decryptable IS NULL)
+				AND hmap.profile_identifier = 'com.fleetdm.fleet.mdm.filevault'
+				AND hmap.status = 'applied'
+				AND hmap.operation_type = 'install') THEN
+			1
+		END) AS action_required, COUNT(
+		CASE WHEN EXISTS (
+			SELECT
+				1 FROM host_mdm_apple_profiles hmap
+			WHERE
+				h.uuid = hmap.host_uuid
+				AND hmap.profile_identifier = 'com.fleetdm.fleet.mdm.filevault'
+				AND hmap.status = 'pending'
+				AND hmap.operation_type = 'install') THEN
```

Comment:
The enforcing status is also returned when the profile is installed (not pending) , the matching hdek row exists BUT the decrytable is NULL. I think this would need a union with the other case (status = applied), it gets complex quite fast I'm not even gonna try to suggest something in this comment box :D ...

Reply:
So I was thinking about this, the other cases all look good to me, so basically the `enforcing` case is the "default" case if none other apply, as long as the host has the filevault profile (in any state except `operation_type='remove' and status = 'applied'`, which is the same as not having the profile). So instead of building a complex query to match the various `enforcing` conditions, you could move it as last `CASE` and just do the check that the host has the filevault profile:

```
-- moving this to the final CASE in the list
CASE WHEN EXISTS (
			SELECT
				1 FROM host_mdm_apple_profiles hmap
			WHERE
				h.uuid = hmap.host_uuid
				AND hmap.profile_identifier = 'com.fleetdm.fleet.mdm.filevault'
				AND NOT (hmap.operation_type = 'remove' AND COALESCE(hmap.status, '') = 'applied')) THEN
			1
		END) AS enforcing
```